imagetest/testworkflow.go

```diff
@@ -225,6 +264,97 @@ func (t *TestWorkflow) addStartStep(stepname, vmname string) (*daisy.Step, error
 }
 
+func (t *TestWorkflow) addCreateNetworkStep(network *daisy.Network) (*daisy.Step, error) {
+	createNetworks := &daisy.CreateNetworks{}
+	*createNetworks = append(*createNetworks, network)
+	createNetworkStep, err := t.wf.NewStep(createNetworkStepName)
+	if err != nil {
+		return nil, err
+	}
+	createNetworkStep.CreateNetworks = createNetworks
+	return createNetworkStep, nil
+}
+
+func (t *TestWorkflow) addCreateSubnetworksStep(subnetwork *daisy.Subnetwork) (*daisy.Step, error) {
+	createSubnetworksStep, err := t.wf.NewStep(createSubNetworkStepName)
+	if err != nil {
+		return nil, err
+	}
+	createSubnetworks := &daisy.CreateSubnetworks{}
+	*createSubnetworks = append(*createSubnetworks, subnetwork)
+	createSubnetworksStep.CreateSubnetworks = createSubnetworks
+	return createSubnetworksStep, nil
+}
+
+// CreateSubNetwork creates custom subnetwork. Using SetCustomNetwork method
+// provided by TestVM to config network on vm
+func (n Network) CreateSubNetwork(name string, ipRange string) (*SubNetwork, error) {
+	subnetwork := &daisy.Subnetwork{
+		Subnetwork: compute.Subnetwork{
+			Name:        name,
+			IpCidrRange: ipRange,
+			Network:     n.name,
+		},
+	}
+	createSubnetworksStep, ok := n.testWorkflow.wf.Steps[createSubNetworkStepName]
+	if !ok {
+		createSubnetworksStep, err := n.testWorkflow.addCreateSubnetworksStep(subnetwork)
+		if err != nil {
+			return nil, err
+		}
+		firstStep, ok := n.testWorkflow.wf.Steps[createDisksStepName]
+		if !ok {
+			return nil, fmt.Errorf("failed resolve first step")
+		}
+		createNetworkStep, ok := n.testWorkflow.wf.Steps[createNetworkStepName]
+		if !ok {
+			return nil, fmt.Errorf("create-network step missing")
+		}
+		if err := n.testWorkflow.wf.AddDependency(createSubnetworksStep, createNetworkStep); err != nil {
+			return nil, err
+		}
+		if err := n.testWorkflow.wf.AddDependency(firstStep, createSubnetworksStep); err != nil {
+			return nil, err
+		}
+		return &SubNetwork{name, n.testWorkflow}, nil
+	}
+
+	*createSubnetworksStep.CreateSubnetworks = append(*createSubnetworksStep.CreateSubnetworks, subnetwork)
+	return &SubNetwork{name, n.testWorkflow}, nil
+}
+
+// CreateNetwork creates custom network. Using SetCustomNetwork method provided by
+// TestVM to config network on vm
+func (t *TestWorkflow) CreateNetwork(networkName string, autoCreateSubnetworks bool) (*Network, error) {
+	network := &daisy.Network{
+		Network: compute.Network{
+			Name: networkName,
+		},
+		AutoCreateSubnetworks: &autoCreateSubnetworks,
+	}
+	createNetworkStep, ok := t.wf.Steps[createNetworkStepName]
+	if !ok {
+		createNetworkStep, err := t.addCreateNetworkStep(network)
+		if err != nil {
+			return nil, err
+		}
+		// Use default subnetwork, no create custom subnetwork step.
+		if autoCreateSubnetworks {
+			firstStep, ok := t.wf.Steps[createDisksStepName]
+			if !ok {
+				return nil, fmt.Errorf("failed resolve first step")
+			}
+			if err := t.wf.AddDependency(firstStep, createNetworkStep); err != nil {
```

Comment:
i think the create-vms step should be the one to depend on create-networks. creating networks and disks can happen in parallel

Reply:
Good points. creating networks and disks and be parallel.